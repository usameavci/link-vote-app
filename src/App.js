import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import LinkContextProvider from "./contexts/LinkContext";
import PageIndex from "./components/pages/Index/Index";
import PageSubmit from "./components/pages/Submit/Submit";
import CommonHeader from "./components/common/Header/Header";
import CommonContainer from "./components/common/Container/Container";

import "./app.css";

const App = () => {
  return (
    <div data-testid="app" className="app">
      <Router>
        <LinkContextProvider>
          <CommonHeader />
          <CommonContainer>
            <Switch>
              <Route path="/submit" component={PageSubmit} />
              <Route path="/" component={PageIndex} />
            </Switch>
          </CommonContainer>
        </LinkContextProvider>
      </Router>
    </div>
  );
};

export default App;
