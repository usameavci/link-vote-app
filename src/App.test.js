import React from "react";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import { create, act as actRenderer } from "react-test-renderer";

import App from "./App";

describe("App component tests", () => {
  it("renders without crashing", async () => {
    const component = <App />;

    let getByTestId;

    await act(async () => ({ getByTestId } = await render(component)));

    expect(getByTestId("app")).toBeInTheDocument();
  });

  it("matches snapshots", async () => {
    const component = <App />;

    let tree;

    await actRenderer(async () => (tree = create(component)));

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
