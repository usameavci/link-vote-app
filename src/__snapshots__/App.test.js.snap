// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests matches snapshots 1`] = `
<div
  className="app"
  data-testid="app"
>
  <div
    className="common-header"
    data-testid="common-header"
  >
    <a
      data-testid="common-logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Hepsiburada"
        className="common-logo"
        src="logo.jpg"
      />
    </a>
    <div
      className="common-header-text"
      data-testid="common-header-text"
    >
      <span>
        Link
      </span>
      <span>
        VOTE
      </span>
      <span>
        Challenge
      </span>
    </div>
  </div>
  <div
    className="common-container"
    data-testid="common-container"
  >
    <div
      className="page-index"
      data-testid="page-index"
    >
      <div
        className="page-index-block"
      >
        <a
          className="link-submit"
          data-testid="link-submit"
          href="/submit"
          onClick={[Function]}
        >
          <div
            className="link-submit-icon"
          >
            <i
              className="fa fa-plus"
            />
          </div>
          <div
            className="link-submit-title"
          >
            SUBMİT A LİNK
          </div>
        </a>
      </div>
      <div
        className="page-index-block"
      >
        <div
          className="link-sort"
          data-testid="link-sort"
        >
          <div
            className="common-select"
            data-testid="common-select"
          >
            <div
              className="common-select-label"
              onClick={[Function]}
            >
              <span
                className="common-select-label-inner"
              >
                Creation Date
              </span>
              <span
                className="common-select-label-icon"
              >
                <i
                  className="fa fa-caret-down"
                />
              </span>
            </div>
            <div
              className="common-select-options"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="common-select-option"
                onClick={[Function]}
              >
                Creation Date
              </div>
              <div
                className="common-select-option"
                onClick={[Function]}
              >
                Most Voted (Z -&gt; A)
              </div>
              <div
                className="common-select-option"
                onClick={[Function]}
              >
                Less Voted (A -&gt; Z)
              </div>
            </div>
          </div>
        </div>
        <div
          className="link-list"
          data-testid="link-list"
        >
          <div
            className="link-item"
            data-testid="link-item"
          >
            <button
              className="link-item-remove"
              data-testid="link-item-remove"
              onClick={[Function]}
            >
              <i
                className="fa fa-times"
              />
            </button>
            <div
              className="link-item-vote"
            >
              <div
                className="vote-point"
              >
                0
              </div>
              <div
                className="vote-label"
              >
                POINTS
              </div>
            </div>
            <div
              className="link-item-body"
            >
              <div
                className="link-item-title"
              >
                Stackowerflow
              </div>
              <div
                className="link-item-url"
              >
                (
                https://stackoverflow.com
                )
              </div>
              <div
                className="link-item-actions"
              >
                <button
                  className="link-item-action"
                  data-testid="link-item-up-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-up"
                  />
                   Up Vote
                </button>
                <button
                  className="link-item-action"
                  data-testid="link-item-down-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-down"
                  />
                   Down Vote
                </button>
              </div>
            </div>
          </div>
          <div
            className="link-item"
            data-testid="link-item"
          >
            <button
              className="link-item-remove"
              data-testid="link-item-remove"
              onClick={[Function]}
            >
              <i
                className="fa fa-times"
              />
            </button>
            <div
              className="link-item-vote"
            >
              <div
                className="vote-point"
              >
                0
              </div>
              <div
                className="vote-label"
              >
                POINTS
              </div>
            </div>
            <div
              className="link-item-body"
            >
              <div
                className="link-item-title"
              >
                Github
              </div>
              <div
                className="link-item-url"
              >
                (
                https://github.com
                )
              </div>
              <div
                className="link-item-actions"
              >
                <button
                  className="link-item-action"
                  data-testid="link-item-up-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-up"
                  />
                   Up Vote
                </button>
                <button
                  className="link-item-action"
                  data-testid="link-item-down-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-down"
                  />
                   Down Vote
                </button>
              </div>
            </div>
          </div>
          <div
            className="link-item"
            data-testid="link-item"
          >
            <button
              className="link-item-remove"
              data-testid="link-item-remove"
              onClick={[Function]}
            >
              <i
                className="fa fa-times"
              />
            </button>
            <div
              className="link-item-vote"
            >
              <div
                className="vote-point"
              >
                0
              </div>
              <div
                className="vote-label"
              >
                POINTS
              </div>
            </div>
            <div
              className="link-item-body"
            >
              <div
                className="link-item-title"
              >
                Linkedin
              </div>
              <div
                className="link-item-url"
              >
                (
                https://linkedin.com
                )
              </div>
              <div
                className="link-item-actions"
              >
                <button
                  className="link-item-action"
                  data-testid="link-item-up-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-up"
                  />
                   Up Vote
                </button>
                <button
                  className="link-item-action"
                  data-testid="link-item-down-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-down"
                  />
                   Down Vote
                </button>
              </div>
            </div>
          </div>
          <div
            className="link-item"
            data-testid="link-item"
          >
            <button
              className="link-item-remove"
              data-testid="link-item-remove"
              onClick={[Function]}
            >
              <i
                className="fa fa-times"
              />
            </button>
            <div
              className="link-item-vote"
            >
              <div
                className="vote-point"
              >
                0
              </div>
              <div
                className="vote-label"
              >
                POINTS
              </div>
            </div>
            <div
              className="link-item-body"
            >
              <div
                className="link-item-title"
              >
                Facebook
              </div>
              <div
                className="link-item-url"
              >
                (
                https://facebook.com
                )
              </div>
              <div
                className="link-item-actions"
              >
                <button
                  className="link-item-action"
                  data-testid="link-item-up-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-up"
                  />
                   Up Vote
                </button>
                <button
                  className="link-item-action"
                  data-testid="link-item-down-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-down"
                  />
                   Down Vote
                </button>
              </div>
            </div>
          </div>
          <div
            className="link-item"
            data-testid="link-item"
          >
            <button
              className="link-item-remove"
              data-testid="link-item-remove"
              onClick={[Function]}
            >
              <i
                className="fa fa-times"
              />
            </button>
            <div
              className="link-item-vote"
            >
              <div
                className="vote-point"
              >
                0
              </div>
              <div
                className="vote-label"
              >
                POINTS
              </div>
            </div>
            <div
              className="link-item-body"
            >
              <div
                className="link-item-title"
              >
                Twitter
              </div>
              <div
                className="link-item-url"
              >
                (
                https://twitter.com
                )
              </div>
              <div
                className="link-item-actions"
              >
                <button
                  className="link-item-action"
                  data-testid="link-item-up-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-up"
                  />
                   Up Vote
                </button>
                <button
                  className="link-item-action"
                  data-testid="link-item-down-vote"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-arrow-down"
                  />
                   Down Vote
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="link-pagination"
          data-testid="link-pagination"
        >
          <div
            className="common-pagination"
            data-testid="common-pagination"
          >
            <div
              className="common-pagination-page common-pagination-page--active"
              data-testid="common-pagination-link"
              onClick={[Function]}
            >
              1
            </div>
            <div
              className="common-pagination-page"
              data-testid="common-pagination-link"
              onClick={[Function]}
            >
              2
            </div>
            <div
              className="common-pagination-page"
              data-testid="common-pagination-link"
              onClick={[Function]}
            >
              <i
                className="fa fa-chevron-right"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
